reverse_engineering/helpers/restApiHelper.js

```diff
@@ -15,30 +21,37 @@ class CouchbaseRestApiService {
 		this.host = connectionInfo.host;
 		this.port = connectionInfo.port;
 
-		if (connectionInfo.couchbase_username && connectionInfo.couchbase_password) {
+		if (connectionInfo?.couchbase_username && connectionInfo?.couchbase_password) {
 			this.password = connectionInfo.couchbase_password;
 			this.username = connectionInfo.couchbase_username;
 		}
 
 		this.httpService = httpService;
 	}
 
-	async fetch(endpoint, options = {}) {
-		const uri = `http://${this.host}:${this.port}${endpoint}`;
+	/**
+	 * @returns {string}
+	 */
+	encodeCredentials() {
+		return Buffer.from(`${this.username}:${this.password}`).toString('base64');
+	}
 
-		if (this.username && this.password) {
-			let encodedCredentials = Buffer.from(`${this.username}:${this.password}`).toString('base64');
-			options = {
-				...options,
+	/**
+	 * @param {string} endpoint
+	 * @returns {Promise<any>}
+	 * @throws {CustomError}
+	 */
+	async fetch(endpoint) {
+		try {
+			const uri = `http://${this.host}:${this.port}${endpoint}`;
+			const encodedCredentials = this.encodeCredentials();
+			const options = {
 				headers: {
-					...(options.headers || {}),
 					Authorization: `Basic ${encodedCredentials}`,
 				},
 				useElectronNet: false,
```

Comment:
Why useElectronNet is disabled?
Auto handling of proxies won't work in this case.